subsys/net/ip/igmp.c

```diff
@@ -326,3 +342,53 @@ int net_ipv4_igmp_leave(struct net_if *iface, const struct in_addr *addr)
 					sizeof(struct in_addr));
 	return ret;
 }
+
+static void net_ipv4_igmp_init(struct net_if *iface)
+{
+	struct net_if_mcast_addr *maddr;
+
+	/* Ensure multicast addresses are available */
+	if (CONFIG_NET_IF_MCAST_IPV4_ADDR_COUNT < 1) {
+		return;
+	}
+
+	/* This code adds the IGMP all systems 224.0.0.1 multicast address
+	 * to the list of multicast addresses of the given interface.
+	 * The address is marked as joined. However, an IGMP membership
+	 * report is not generated for this address. Populating this
+	 * address in the list of multicast addresses of the interface
+	 * and marking it as joined is helpful for multicast hash filter
+	 * implementations that need a list of multicast addresses it needs
+	 * to add to the multicast hash filter after a multicast address
+	 * has been removed from the membership list.
+	 */
+	maddr = net_if_ipv4_maddr_lookup(&all_systems, &iface);
+	if (maddr && net_if_ipv4_maddr_is_joined(maddr)) {
+		return;
+	}
+
+	if (!maddr) {
+		maddr = net_if_ipv4_maddr_add(iface, &all_systems);
+		if (!maddr) {
+			return;
+		}
+	}
+
+	net_if_ipv4_maddr_join(maddr);
+
+	net_if_mcast_monitor(iface, &maddr->address, true);
+}
+
+void init_igmp(struct net_if *iface)
+{
+#if defined(CONFIG_NET_IPV4_IGMP)
```

Comment:
This seems redundant, if `CONFIG_NET_IPV4_IGMP` is disabled the file will not be compiled.

Reply:
Thanks. I think I had misunderstood @jukkar's comment from before. I've moved this function out of igmp.c and into net_if.c where `CONFIG_NET_IPV4_IGMP` would be relevant. Its incorporated in my latest change.